python/legate_dataframe/lib/core/scalar.pyx

```diff
@@ -40,6 +41,9 @@ cdef LogicalColumn cpp_scalar_col_from_python(scalar: ScalarLike):
     if isinstance(scalar, legate.core.Scalar):
         scalar = scalar.value()
 
+    if isinstance(scalar, pa.Scalar):
+        return LogicalColumn.from_arrow(scalar)
```

Comment:
Maybe remove the typing here, but add `pa.Scalar` to the `.pyi`.  (My bad for duplicating it here.)

Reply:
The `scalar.pyi` defines `ScalarLike`.  It makes no sense to keep `ScalarLike` around here, but it makes sense to include `pa.Scalar` there.